File: rot13/cpp/rot13.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <map>
#include <iterator>
using namespace std;

string rot13(string text, map<char, char> mapping) {
    string cipher_shift = "";
    for (int i = 0; i < text.length(); i++) {
        if (mapping.count(text[i])) cipher_shift += mapping[text[i]];
        else cipher_shift += text[i];
    }
    return cipher_shift;
}

void test_cipher() {
    // check if encryption works
    ifstream file_1("../OGtext.txt");
    string text_1((std::istreambuf_iterator<char>(file_1)),
                (std::istreambuf_iterator<char>()));
    file_1.close();
    ifstream file_2("../decrypted.txt");
    string text_2((std::istreambuf_iterator<char>(file_2)),
                (std::istreambuf_iterator<char>()));
    file_2.close();
    if (text_1 == text_2) cout << true << '\n';
    else cout << false << '\n';

    cout << text_1.length() << '\n';
    cout << text_2.length() << '\n';
}

int main() {
    char choice;
    while (choice != 'e' and choice != 'd') {
        cout << "Want to encrypt or decrypt your cipher? Input e/d: ";
        cin >> choice;
    }

    map<char, char> mapping;
    for (int i = 0; i < 26; i++) {
        mapping.insert(make_pair<char, char> ('a' + i, ((i + 13) % 26) + 'a'));
        mapping.insert(make_pair<char, char> ('A' + i, ((i + 13) % 26) + 'A'));
    }

    if (choice == 'e') {
        ifstream ifs("../OGtext.txt");
        string text((std::istreambuf_iterator<char>(ifs)),
                    (std::istreambuf_iterator<char>()));
        ifs.close();
        ofstream encrypted("../encrypted.txt");
        encrypted << rot13(text, mapping);
        encrypted.close();
    }

    if (choice == 'd') {
        ifstream ifs("../encrypted.txt");
        string text((std::istreambuf_iterator<char>(ifs)),
                    (std::istreambuf_iterator<char>()));
        ifs.close();
        ofstream decrypted("../decrypted.txt");
        decrypted << rot13(text, mapping);
        decrypted.close();
    }
    
    test_cipher();
}

File: rot13/python/rot13.py
import os.path
import argparse

def rot13(text, mapping):
    cipher_shift = ""
    for i in range(len(text)):
        cipher_shift += mapping.get(text[i], text[i])
    return cipher_shift

def main():
    mapping = {}
    for i in range(26):
        mapping[chr(i + ord('a'))] = chr((i + 13) % 26 + ord('a'))
        mapping[chr(i + ord('A'))] = chr((i + 13) % 26 + ord('A'))
    
    text_path = os.path.dirname(os.getcwd())
    parser = argparse.ArgumentParser()
    parser.add_argument("--choice")
    args = parser.parse_args()

    if args.choice == 'e':
        file = open(text_path + '/OGtext.txt')
        text = file.read()

        e_text = open(text_path + "/encrypted.txt", 'w')
        e_text.write(rot13(text, mapping))
        e_text.close()

    if args.choice == 'd':
        file = open(text_path + '/encrypted.txt')
        text = file.read()

        d_text = open(text_path + "/decrypted.txt", 'w')
        d_text.write(rot13(text, mapping))
        d_text.close()

main()